src/api/transformers/ContactProfileTransformer.ts

```diff
@@ -0,0 +1,39 @@
+import { GetContactProfileDto } from "@api/dto/ContactProfileDto";
+import AddressTransformer from "@api/transformers/AddressTransformer";
+import { BaseTransformer } from "@api/transformers/BaseTransformer";
+
+import Contact from "@models/Contact";
+import ContactProfile from "@models/ContactProfile";
+
+class ContactProfileTransformer extends BaseTransformer<
+  ContactProfile,
+  GetContactProfileDto
+> {
+  protected model = ContactProfile;
+  protected filters = {};
+
+  convert(
+    profile: ContactProfile,
+    opts: unknown,
+    caller: Contact | undefined
+  ): GetContactProfileDto {
+    return {
+      telephone: profile.telephone,
+      twitter: profile.twitter,
+      preferredContact: profile.preferredContact,
+      deliveryOptIn: profile.deliveryOptIn,
+      deliveryAddress:
+        profile.deliveryAddress &&
+        AddressTransformer.convert(profile.deliveryAddress),
+      newsletterStatus: profile.newsletterStatus,
+      newsletterGroups: profile.newsletterGroups,
+      ...(caller?.hasRole("admin") && {
```

Comment:
Ah is this replacing `opts.withRestricted`?

Reply:
Yup, trying to put all permissions based logic within the transformers so you can't accidentally convert with the restricted flag.